packages/babel-preset-expo/src/inline-env-vars.ts

```diff
@@ -43,3 +43,43 @@ export function expoInlineEnvVars(api: ConfigAPI & { types: typeof types }): Plu
     },
   };
 }
+
+/**
+ * Given a set of options like `{ EXPO_BASE_URL: '/' }`, inline the values into the bundle.
+ * This is used for build settings that are always available and not configurable at runtime.
+ *
+ * Webpack uses DefinePlugin for similar functionality.
+ */
+export function expoInlineTransformEnvVars(api: ConfigAPI & { types: typeof types }): PluginObj {
+  const { types: t } = api;
+
+  function isFirstInAssign(path: NodePath<types.MemberExpression>) {
+    return t.isAssignmentExpression(path.parent) && path.parent.left === path.node;
+  }
+
+  return {
+    name: 'expo-inline-transform-environment-variables',
+    visitor: {
+      MemberExpression(path, state) {
+        const options = state.opts as Record<string, string>;
+
+        if (path.get('object').matchesPattern('process.env')) {
+          // @ts-expect-error: missing types
+          const key = path.toComputedKey();
+          if (
+            t.isStringLiteral(key) &&
+            !isFirstInAssign(path) &&
+            options[key.value] !== undefined
+          ) {
+            debug(
+              'Inlining transform setting in %s: %s',
+              state.filename || '[unknown file]',
```

Comment:
```suggestion
              state.filename ?? '[unknown file]',
```

Reply:
This is a debug log, I'd rather have something than nothing.

Reply:
Could you explain more about why this would be nothing?